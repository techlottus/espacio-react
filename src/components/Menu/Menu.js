import React, { useEffect } from "react";



export const Menu = React.memo(({ data, onClose, onCard,outside }) => {
  const menuRef = React.createRef();

  useEffect(() => {
    menuRef.current.data = {
      items: data.items || [],
      disabledBtnClose: data.disabledBtnClose,
      isShow: data.isShow,
    };
  },[data]) // eslint-disable-line react-hooks/exhaustive-deps


  useEffect(() => {
    menuRef.current.addEventListener("onClose", onClose);
    menuRef.current.addEventListener("onCard", onCard);
    const handleClickOutside = (event) => {
      if (menuRef.current && !menuRef.current.contains(event.target) && window.innerWidth > 991) {
        outside();
      }
      
    }
    document.addEventListener("mousedown", handleClickOutside);
    return () => {
      document.removeEventListener("mousedown", handleClickOutside);
    };
  }, []); // eslint-disable-line react-hooks/exhaustive-deps

  return (
    <>
      <lottus-menu ref={menuRef}></lottus-menu>
    </>
  );
});
