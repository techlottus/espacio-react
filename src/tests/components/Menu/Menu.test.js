import React, { createRef } from "react";
import { mount } from "enzyme";
import { Provider } from "react-redux";
import "@testing-library/jest-dom";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import { Menu } from "../../../components/Menu/Menu";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

let initState;
let data = {
  items: null,
  disabledBtnClose: false,
  isShow: true,
};
let onClose = jest.fn();
let onCard = jest.fn();
let outside = jest.fn();
let menuRef = "";

let store = mockStore(initState);
const handleMount = (initState) => {
  store = mockStore(initState);
  return mount(
    <Provider store={store}>
      <Menu
        data={data}
        onClose={onClose}
        onCard={onCard}
        outside={outside}
        menuRef={menuRef}
      />
    </Provider>
  );
};

describe("Test <Menu/>", () => {
  test("should init component", () => {
    const wrapper = handleMount(initState);
    expect(wrapper).toMatchSnapshot();
    wrapper.unmount();
  });

  // test("should put window to 991", () => {
  //   initState = {
  //     menuRef: {
  //       ...initState.menuRef,
  //     },
  //   };
  //   window.innerWidth = 991;
  //   const wrapper = handleMount(initState);
  //   wrapper.update();
  // });
});
