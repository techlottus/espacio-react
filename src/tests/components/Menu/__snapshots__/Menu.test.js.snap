// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Menu/> should init component 1`] = `<lottus-menu />`;
